### PR TITLE
Migrate sync job to GitHub Actions and remove CircleCI config

### LINKED ISSUE #2968: Move the `main` -> `develop` sync job to GitHub Actions
## Description
Part of the migration to GHA

## Context

Currently done by CircleCi 
https://github.com/kedro-org/kedro/blob/e2bb4cd5cac9e25248046f2be2015f734b8d8d37/.circleci/continue_config.yml#L598

### LINKED ISSUE #3318: Remove CircleCI config completely
## Description
After [Remove CircleCI Tests from config#2972](https://github.com/kedro-org/kedro/issues/2972) the only remaining workflows are:

- [x] https://github.com/kedro-org/kedro/issues/2968
- [x] https://github.com/kedro-org/kedro/issues/2969
- [ ] Remove the CCI jobs from `kedro-starters`

## Context
This is part of GHA migration so we need to remove the CircleCI config once the remaining work above has been merged.

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description
Fix #2968 
Also fix #3318 

### `sync.yml`

The first part of the workflow is to merge the commits to `develop` if they can be merged cleanly otherwise open a PR.
This workflow uses the same script that we did with CircleCI except it's triggered by GHA instead. This workflow runs on a schedule (every 30 mins) but can also be dispatched manually.

🔔  Workflow in action on my fork - https://github.com/ankatiyar/kedro/actions/workflows/sync.yml

 
### `.github/workflows/label-pr.yml` & `.github/labeler.yml`

This second workflow uses the [labeler action](https://github.com/actions/labeler). The workflow is in `.github/workflows/label-pr.yml` and the configuration is in `.github/labeler.yml`. This workflow labels any PRs on the branch `merge-main-to-develop` with the "automerge" label which is required by the auto-merge action described below.
NOTE - this action can be extended in the future for auto labelling of other PRs such as "Community" etc

### `auto-merge-prs.yml`

This action is the final part of the workflow - it merges any PRs labelled "automerge" with a merge commit. It uses the [auto-merge action](https://github.com/pascalgn/automerge-action). This is also a scheduled action (every 60 mins) and can also be triggered manually.

🔔 https://github.com/ankatiyar/kedro/pull/14 - PR created, labelled and merged with the above workflows on my fork

### Removal of all CircleCI stuff 
Since it was the only workflow left and wouldn't make sense to let it run on both platforms incase there are conflicts with merging.

NOTE - I wasn't able to get the second script we run used to run with CircleCI `tools/circleci/github_scripts/attempt-merge-pr.sh` so this is an alternative solution for the merging of PRs with marketplace actions. 

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by merelcht — APPROVED** [review]
This looks really good! Great job @ankatiyar ⭐ 👍  I am in favour of using the GH actions marketplace where we can. The only question I have is whether it's possible to add a slightly longer description in the workflow .yml files to explain what they do? Otherwise I think it would be a good idea to document in our wiki what each of these jobs do.

File: .circleci/config.yml
Comment (SajidAlamQB) [inline .circleci/config.yml:1]: No more CircleCI! 🥳

**Review by SajidAlamQB — APPROVED** [review]
Awesome work @ankatiyar! 🌟 The two new GH actions make sense and look to be working well. Excited to see this in action!

**Comment by ankatiyar** [comment]
@merelcht I've added comments to the workflows but I'll also create a wiki for the CI setup in general.

**Pull request merged**